client/finality-grandpa/src/lib.rs

```diff
@@ -670,6 +675,7 @@ where
 			authority_set: persistent_data.authority_set.clone(),
 			consensus_changes: persistent_data.consensus_changes.clone(),
 			voter_set_state: persistent_data.set_state.clone(),
+			metrics: prometheus_registry.map(|registry| Metrics::register(&registry).unwrap()),
```

Comment:
At least `.expect("Reason/what failed")` instead of `unwrap`

And are we. sure we can panic here at all?

Reply:
> And are we. sure we can panic here at all?

Pretty sure we can't. One should be able to operate finality-grandpa without a Prometheus registry, in which case this would panic.

Reply:
No, I don't believe we can panic here because we try to register other metrics before this.